File: day9/binarySearch.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        ld = num%10;  #ld- last digit because here num is square not square root
        if(ld ==2 or ld == 3 or ld == 7 or ld == 8 ):
            return False
        
        #binary search
        
        l=1
        r = num
        while(l<= r):
            mid =int(l +(r-l)/2)
            sq = int(mid*mid)
            
            if(sq ==num):
                return True
            elif(sq< num):
                l = mid+1
            else:
                r = mid-1        
        
        return False
        
    

File: day9/binarySearch.java
class Solution {
    public boolean isPerfectSquare(int num) {
         int ld = num%10;    //ld- last digit because here num is square not square root
        if(ld ==2 || ld == 3 || ld == 7 || ld == 8 )
            return false;
        
        //binary search
        
        int l=1;
        int r = num;
        while(l<= r){
            long mid = l +(r-l)/2;
            long sq = mid*mid;
            
            if(sq ==num)
                return true;
            else if(sq< num)
                l = (int)mid+1;
            else
                r = (int)mid-1;        
        }
        return false;
        
    }
}